File: Lab8/C#/ClassLib/MyMath.cs
using System;

namespace ClassLib
{
    public class MyMath
    {
        public delegate double Operation(double value1, double value2);
        

        public double DoMath(double x, double y, Operation op)
        {
            double res = 0;
            if (op != null)
                res = op(x, y);
            return res;            
        }

        static public double Add(double x, double y) => x + y;
        static public double Subtract(double x, double y) => x - y;
        static public double Multiply(double x, double y) => x * y;

        static public double Divide(double x, double y) 
        {
            if (y == 0)
                throw new DivideByZeroException("Division by zero");
            return x / y;
        }
        public double MathAdd(double x, double y) => x + y;
    }
}

File: Lab8/C++/lab8/MyMath.cpp
#include "MyMath.h"
double MyMath::DoOperation(double value1, double value2, double(MyMath::* operation)(double value1, double value2))
{
	if (operation == nullptr)
		throw new std::exception("Operation must not be null");
	return (this->*operation)(value1, value2);
}
double MyMath::Add(double value1, double value2)
{
	return value1 + value2;
}
double MyMath::Subtract(double value1, double value2)
{
	return value1 - value2;
}
double MyMath::Multiply(double value1, double value2)
{
	return value1 * value2;
}
double MyMath::Divide(double value1, double value2)
{
	if (value2 == 0)
		throw new std::exception("Value2 must not be 0");
	return value1 / value2;
}